### PR TITLE
Remove tagging from AddCommand

### PR DESCRIPTION
**Pull request by Kalsyc**
Remove tagging from AddCommand
Edit some test cases
Modify TagCommand such that it only accepts arguments with hex as tags

TODO:
Implement delete tag
Edit UserGuide
Create tests for the new implementation of TagCommand and AddCommand

### CODE REVIEW

File: src/main/java/io/xpire/logic/commands/TagCommand.java [hunk 1]
```diff
@@ -3,33 +3,33 @@
...
 public class TagCommand extends Command {
 
     public static final String COMMAND_WORD = "tag";
 
     public static final String MESSAGE_USAGE = COMMAND_WORD
             + ": Tags the item identified by the index number used in the displayed item list.\n"
-            + "Parameters: INDEX (must be a positive integer)\n"
-            + "|#TAG... (if missing, item will be cleared of its tags)\n"
-            + "Example: " + COMMAND_WORD + "|1|#Food";
+            + "Parameters: <index> (must be a positive integer)\n"
```
Comment (JermyTan) [inline src/main/java/io/xpire/logic/commands/TagCommand.java]: Should be replaced with "Format:...."

File: src/test/data/JsonSerializableXpireTest/typicalItemsXpire.json [hunk 2]
```diff
@@ -3,17 +3,14 @@
   "items" : [ {
     "name" : "Milk",
     "expiryDate" : "01/02/2019",
-    "reminderThreshold" : "0",
-    "tags" : [ "drinks" ]
```
Comment (JermyTan) [inline src/test/data/JsonSerializableXpireTest/typicalItemsXpire.json]: JSON item model is
{
    "name: <name>
    "expireDate" : <date>
    "reminderThreshold" : <threshold>
    "tags" : <array of tags>
}

so there should be a field "tags"

**Comment by Kalsyc** [comment]
Modify tagging fuctionality such that it parses tags in sentence-case
Fix ResultDisplay such that it only displays tags if there are tags
Edit UserGuide to reflect current Delete/Tag/Add commands
Shift deletion of tags to DeleteCommand

To do:
Fix J-Unit testing (will be done after quantity has been added in)

**Comment by febee99** [comment]
LGTM 👍🏻

File: src/main/java/io/xpire/logic/commands/DeleteCommand.java [hunk 3]
```diff
@@ -3,31 +3,58 @@
...
 import io.xpire.commons.core.index.Index;
 import io.xpire.logic.commands.exceptions.CommandException;
 import io.xpire.model.Model;
 import io.xpire.model.item.Item;
+import io.xpire.model.tag.Tag;
+import io.xpire.model.tag.TagComparator;
 
 /**
- * Deletes an item identified with its displayed index.
+ * Deletes an item identified with its displayed index or tag(s) associated with the item.
```
Comment (xiaoyu-nus) [inline src/main/java/io/xpire/logic/commands/DeleteCommand.java]: if delete can not only delete items but also quality, should i also enable deleting reminder thru the command delete?

File: src/main/java/io/xpire/logic/commands/TagCommand.java [hunk 4]
```diff
@@ -111,12 +120,38 @@ public TagItemDescriptor(TagItemDescriptor toCopy) {
...
+            // instanceof handles nulls
+            if (!(other instanceof TagItemDescriptor)) {
+                return false;
+            }
+
+            // state check
+            TagItemDescriptor e = (TagItemDescriptor) other;
+
+            return getTags().equals(e.getTags());
+        }
```
Comment (xiaoyu-nus) [inline src/main/java/io/xpire/logic/commands/TagCommand.java]: the .equals method format is different from DeleteCommand.java. Are we fixing on one format?

Comment (Kalsyc) [reply src/main/java/io/xpire/logic/commands/TagCommand.java]: I kinda based the TagCommand equals method on the EditCommand. 
For DeleteCommand I'm not very sure what should be the equal method. I think it should depend on whether we are deleting item/quantity/tags.

File: src/main/java/io/xpire/logic/commands/DeleteCommand.java (hunk 3 above)
Comment (Kalsyc) [reply src/main/java/io/xpire/logic/commands/DeleteCommand.java]: Hmm... Would we need to delete reminders for items? Can we just set it back to 0 as the default

**Pull request merged**